php/blocks/controls/class-image.php

```diff
@@ -46,15 +53,26 @@ public function register_settings() {
 				'sanitize' => 'sanitize_text_field',
 			)
 		);
-		$this->settings[] = new Control_Setting(
-			array(
-				'name'     => 'default',
-				'label'    => __( 'Default Value', 'block-lab' ),
-				'type'     => 'url',
-				'default'  => '',
-				'sanitize' => 'esc_url_raw',
-				'help'     => __( 'An image URL.' ),
-			)
-		);
+	}
+
+	/**
+	 * Validates the value to be made available to the front-end template.
+	 *
+	 * @param string $value The value to either make available as a variable or echoed on the front-end template.
+	 * @param bool   $echo Whether this value will be echoed.
+	 * @return string|int|false $value The value to be made available or echoed on the front-end template.
+	 */
+	public function validate( $value, $echo ) {
+		// Backwards compatibility, as the value used to be the image's URL instead of its post ID.
+		if ( empty( intval( $value ) ) ) {
+			$value = attachment_url_to_postid( $value );
```

Comment:
NICE! This means there's no back-compat break, right?

Reply:
Yeah, this should minimize the back-compat break. If someone just entered an external URL in the `<input>`, this wouldn't help. But it'll usually find the ID for a URL that's from the Media Library.

And as you know, the output of `block_value( 'foo-image-field' )` is changing to output an ID instead of the URL string.

Reply:
[Below](https://github.com/getblocklab/block-lab/pull/260#issuecomment-479475773) are more details on backwards-compatibility.